service/src/main/java/bio/terra/cbas/common/MicrometerMetrics.java

```diff
@@ -15,27 +23,94 @@ public MicrometerMetrics(MeterRegistry meterRegistry) {
     this.meterRegistry = meterRegistry;
   }
 
-  public void logRunCallback(CbasRunStatus resultsStatus) {
-    Counter counter =
-        Counter.builder("run_callback")
-            .tag("status", resultsStatus.toString())
-            .register(meterRegistry);
-    counter.increment();
+  /* ******** Counter metrics ******** */
+
+  public void recordRunCallback(CbasRunStatus resultsStatus) {
+    recordCounterMetric(
+        "run_callback", List.of(new ImmutableTag("status", resultsStatus.toString())), 1);
+  }
+
+  public void recordRunStatusUpdate(CbasRunStatus resultsStatus) {
+    recordCounterMetric(
+        "run_smartpoller_update", List.of(new ImmutableTag("status", resultsStatus.toString())), 1);
+  }
+
+  public void increaseEventCounter(String metricName, long count) {
+    recordCounterMetric(metricName, Collections.emptyList(), count);
+  }
+
+  /* ******** Event distribution metrics ******** */
+
+  public void recordEventDistributionMetric(String metricName, int count) {
+    recordEventDistributionMetric(metricName, Collections.emptyList(), count);
+  }
+
+  public void recordRunsSubmittedPerRunSet(UUID runSetId, long runsCount) {
+    recordEventDistributionMetric(
+        "runs_submitted_successfully_per_run_set",
+        List.of(new ImmutableTag("run_set_id", runSetId.toString())),
+        runsCount);
+  }
+
+  /* ******** Timer metrics ******** */
+
+  public void recordPostMethodHandlerCompletion(
+      Timer.Sample sample, String source, int responseCode) {
+    recordTimerMetric(
+        sample,
+        "post_method_response_timing",
+        List.of(
+            new ImmutableTag("source", source),
+            new ImmutableTag("response_code", String.valueOf(responseCode))));
+  }
+
+  public void recordPostRunSetHandlerCompletion(
+      Timer.Sample sample, int inputsCount, int outputsCount, int recordsCount, int responseCode) {
+    recordTimerMetric(
+        sample,
+        "post_runset_response_timing",
+        List.of(
+            new ImmutableTag("inputs_count", String.valueOf(inputsCount)),
+            new ImmutableTag("outputs_count", String.valueOf(outputsCount)),
+            new ImmutableTag("records_count", String.valueOf(recordsCount)),
+            new ImmutableTag("response_code", String.valueOf(responseCode))));
+  }
```

Comment:
This is a new metric that I added while I was here as it relates to submission scaling. The purpose of adding this metric is to see the response timing of the POST /run_set API and benchmark submit performance when we make more improvements in the future.